### PR TITLE
Add CategoryRule documentation

### PR DESCRIPTION
**Pull request by Concetta**
This PR adds CategoryRule endpoints to the OpenAPI as follows: 
* GET  List category rules in user
* POST Create category rule in category

# Checklist

- [x] The OpenAPI JSON file was produced and formatted using Swagger Editor
- [x] The OpenAPI JSON file validates with no errors or warnings in Swagger Editor
- [x] This is a public repo. Your branch name, commits and pull request are appropriate and don't reference internal tooling

### CODE REVIEW

File: openapi.json [hunk 1]
```diff
@@ -2599,4 +2750,4 @@
       "developerKey": []
     }
   ]
-}
+}
```
Comment (davisonja) [inline openapi.json]: For completeness: new line here :)

File: openapi.json [hunk 2]
```diff
@@ -2363,6 +2395,125 @@
...
+                "required": [
+                  "category_id",
+                  "payee_matches"
+                ],
+                "properties": {
+                  "category_id": {
+                    "type": "integer",
+                    "description": "The ID of the category the rule is for.",
+                    "example": 42
+                  },
```
Comment (davisonja) [inline openapi.json]: The category id is already in the path, our other endpoints don't duplicate it.

File: openapi.json [hunk 3]
```diff
@@ -2363,6 +2395,125 @@
...
+                  "payee_matches": {
+                    "type": "string",
+                    "description": "The keyword/s to match the transaction payees",
+                    "example": "Countdown"
+                  },
+                  "apply_to_uncategorised": {
+                    "type": "boolean",
+                    "description": "Apply the category to all uncategorised transactions matching the keyword/s.",
```
Comment (davisonja) [inline openapi.json]: This specifically applies **only** the newly created rule, not all the rules that the category has (as there may have already been some). 

So maybe
`Apply the created category rule...`
highlights that?

File: openapi.json [hunk 4]
```diff
@@ -2363,6 +2395,125 @@
...
+                    "example": true
+                  },
+                  "apply_to_all": {
+                    "type": "boolean",
+                    "description": "Apply the category to all transactions matching the keyword/s.",
```
Comment (davisonja) [inline openapi.json]: Same here - this specifically applies **only** the newly created rule, not all the rules that the category has.

**Review by davisonja — CHANGES_REQUESTED** [review]
Two changes, and a phrasing query, otherwise looking 👍

File: openapi.json (hunk 3 above)
Comment (Concetta) [reply openapi.json]: That does sound better 💐

**Review by davisonja — APPROVED** [review]

**Pull request merged**